sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/ResultParameterProcessor.java

```diff
@@ -0,0 +1,761 @@
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.sql.parameterprocessor;
+
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.StructureType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.XmlUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BXml;
+import org.ballerinalang.sql.Constants;
+import org.ballerinalang.sql.exception.ApplicationError;
+import org.ballerinalang.sql.utils.ColumnDefinition;
+import org.ballerinalang.sql.utils.ErrorGenerator;
+import org.ballerinalang.sql.utils.ModuleUtils;
+import org.ballerinalang.sql.utils.Utils;
+
+import java.math.BigDecimal;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+import static io.ballerina.runtime.api.utils.StringUtils.fromString;
+
+/**
+ * This class implements methods required convert SQL types into ballerina types and
+ * other methods that process the parameters of the result.
+ *
+ * @since 0.5.6
+ */
+public class ResultParameterProcessor extends AbstractResultParameterProcessor {
+    private static final Object lock = new Object();
+    private static volatile ResultParameterProcessor instance;
+
+    private static final ArrayType stringArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE_STRING);
+    private static final ArrayType booleanArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE_BOOLEAN);
+    private static final ArrayType intArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE_INT);
+    private static final ArrayType floatArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE_FLOAT);
+    private static final ArrayType decimalArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE_DECIMAL);
+
+    private static final Calendar calendar = Calendar
+            .getInstance(TimeZone.getTimeZone(Constants.TIMEZONE_UTC.getValue()));
+
+
+    public static ResultParameterProcessor getInstance() {
```

Comment:
Instead of using instances can we use static methods  here?  For ex:

```public class AbstractResultParameterProcessor {
    public static Object convert (….) throws ApplicationError {
        throw ApplicationError(“Method not implemented”)
    }
}

public class ResultParameterProcessor extends AbstractResultParameterProcessor {
    public static Object convert (….) throws ApplicationError {
       //Real logic
    }
}


//Call the methods
ResultParameterProcessor.convert(….);```

Reply:
We decided to go with non-static methods of instances, instead of static methods because we need these methods to be overrriden in the connector modules to have connector specific logic. But it is not possible to override static methods.